### PR TITLE
Rework FIWARE interoperability section

### PR DESCRIPTION
**Pull request by jamoralp**

### CODE REVIEW

File: _docs/overview/fiware_interoperability/index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: Interoperability with FIWARE
+permalink: /docs/overview/fiware_interoperability/
+---
+
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:5]: I'd put a pair of lines before explaining what the overall section is about, and to give a little bit of context, such as: "Among the goals of the micro-ROS projects, one of the key issues has been that of providing interoperability with third outstanding and broadly used platforms. One of the selected components has been the FIWARE Context Broker, a standard for context data management adopted by several EU boosted initiatives for facilitating the development of smart solutions for different domains. This section explains how to achieve interoperability between micro-ROS and this platform, passing through the integration of the latter with ROS 2. Thanks to this interoperability, the Context Broker becomes micro-ROS' platform of choice for sharing context information with any other system integrated into the FIWARE ecosystem. "

File: _docs/overview/fiware_interoperability/index.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+## Interoperability: pros and cons of the different possible solutions
+
+This subsection will explain all the design alternatives for the interoperability between micro-ROS and the FIWARE Context Broker.
+From now on, the developed solution for intercommunicating micro-ROS with FIWARE will be called **FIROS2 integration service**.
+
+FIROS2 requires transformation libraries to convert ROS 2 messages into FIWARE NGSIv2 messages and the other way around.
+For each message, one transformation library is required by the integration service (FIROS2).
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:12]:
```suggestion
For each message, one transformation library is required.
```

File: _docs/overview/fiware_interoperability/index.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+---
+title: Interoperability with FIWARE
+permalink: /docs/overview/fiware_interoperability/
+---
+
+## Interoperability: pros and cons of the different possible solutions
+
+This subsection will explain all the design alternatives for the interoperability between micro-ROS and the FIWARE Context Broker.
+From now on, the developed solution for intercommunicating micro-ROS with FIWARE will be called **FIROS2 integration service**.
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:9]:
```suggestion
From now on, the developed solution for intercommunicating micro-ROS with FIWARE will be called **FIROS2 Integration Service**.
```

File: _docs/overview/fiware_interoperability/index.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+* Implementing an ad-hoc bridging communication tool for translating FIWARE's messages into micro-ROS (that is, ROS 2) messages types, and viceversa.
+* Relying on an integration platform that uses a common types language representation, and defines a conversion library from/to the generic type to the specific type of each middleware.
+
+While the first approach might result in a more lightweight tool, it has several flaws, for instance a more difficult maintenance and the incapability of communicating with any other middleware, rather than ROS2 or micro-ROS.
+On the other hand, using an integration service platform, such as [SOSS](https://github.com/eProsima/soss_v2), enables automatically the possibility of communicating with a wide (and growing) set of middlewares, if their System Handle implementation is available.
+
+## SOSS: System Of Systems Synthesizer
+
+**SOSS** addresses the task of providing a common interface for communicating software platforms that speak different languages.
+It is composed of a **core** library, which defines set of abstract interfaces and provides some utility classes to form a plugin-based framework.
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:32]:
```suggestion
It is composed of a **core** library, which defines a set of abstract interfaces and provides some utility classes to form a plugin-based framework.
```

File: _docs/overview/fiware_interoperability/index.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+This pluggable interface allows the user to use one of the supported plugins or System Handles for a specific middleware, such as DDS, ROS2, FIWARE or ROS.
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:34]:
```suggestion
This pluggable interface allows the user to leverage any of the supported plugins or System Handles for a specific middleware, such as DDS, ROS2, FIWARE, or ROS, for the desired integration.
```

File: _docs/overview/fiware_interoperability/index.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+Usually, types are defined using a common language representation, used by SOSS to create a shared representation of the exchanged information, so that it can be processed, converted and remapped to every middleware's types implementation, when required.
+This common representation is provided, user-wise, using IDL definitions, which are parsed and converted into Dynamic Types representations at runtime, using [eProsima's XTypes-DDS](https://github.com/eProsima/xtypes) implementation.
+
+<img width="600" src="imgs/soss.png">
+
+## SOSS-FIWARE system handle
+
+The [FIWARE System Handle](https://github.com/eProsima/SOSS-FIWARE/tree/feature/xtypes-support) allows bringing information from and to FIWARE's Context Broker into the SOSS world.
+This [System Handle](https://soss.docs.eprosima.com/en/latest/sh_creation.html) is configured and launched the same way as any SOSS System Handle.
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:49]:
```suggestion
This [System Handle](https://soss.docs.eprosima.com/en/latest/sh_creation.html) is configured and launched in the same way as any SOSS System Handle.
```

File: _docs/overview/fiware_interoperability/index.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+Besides the standard information included in any System Handle's configuration (such as system's name and type, which would be fiware for this specific System Handle),
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:51]:
```suggestion
Besides the standard information included in any System Handle's configuration (such as system's name and type, which would be `fiware` for this specific System Handle),
```

File: _docs/overview/fiware_interoperability/index.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+in the case of the FIWARE System Handle users must specify two extra YAML key-value pairs, which are the host's IP and port in which this System Handle will try to connect to an instance of FIWARE's Orion Context Broker.
+
+Regarding more specific details about the implementation, FIWARE does not allow certain characters in its entities names.
+For this reason, if a type defined in the topics section of the configuration file has in its name a /, the FIWARE System Handle will map that character into two underscores.
+This is something important to notice when connecting to ROS2, because in ROS2 most of the types have a / in their names.
+To deal with this issue, using SOSS [remapping](https://soss.docs.eprosima.com/en/latest/yaml_config.html?highlight=remap#remapping) capabilities come in handy.
+
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:58]: Before starting a new subsection, I'd remark that the final integration we're aiming at -between fiware and micro-ROS- passes through ROS 2, and in this respect say something about the ROS 2 System Handle, so as to make the description more symmetric.

**Review by FranFin — CHANGES_REQUESTED** [review]

File: _docs/overview/fiware_interoperability/index.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+Besides the standard information included in any System Handle's configuration (such as system's name and type, which would be `fiware` for this specific System Handle),
+in the case of the FIWARE System Handle users must specify two extra YAML key-value pairs, which are the host's IP and port in which this System Handle will try to connect to an instance of FIWARE's Orion Context Broker.
+
+Regarding more specific details about the implementation, FIWARE does not allow certain characters in its entities names.
+For this reason, if a type defined in the topics section of the configuration file has in its name a /, the FIWARE System Handle will map that character into two underscores.
+This is something important to notice when connecting to ROS2, because in ROS2 most of the types have a / in their names.
+To deal with this issue, using SOSS [remapping](https://soss.docs.eprosima.com/en/latest/yaml_config.html?highlight=remap#remapping) capabilities come in handy.
+
+Of course, given that micro-ROS applications act as a bridge between microcontrollers and the ROS 2 dataspace (using the micro-ROS Agent), FIROS2 should take care of communicating FIWARE's Context Broker with ROS 2.
```
Comment (FranFin) [inline _docs/overview/fiware_interoperability/index.md:66]:
```suggestion
Of course, given that micro-ROS applications act as a bridge between microcontrollers and the ROS 2 dataspace (using the micro-ROS Agent), FIROS2 should take care of also communicating FIWARE's Context Broker with ROS 2, leveraging the existing [ROS 2 System Handle](https://github.com/eProsima/soss_v2/tree/master/packages/ros2), which comes natively included into the SOSS package.
```

**Review by FranFin — CHANGES_REQUESTED** [review]

**Review by FranFin — APPROVED** [review]

**Review by pablogs9 — APPROVED** [review]

**Review by ralph-lange — APPROVED** [review]

**Pull request merged**